src/scheduler.rs

```diff
@@ -88,6 +93,21 @@ impl Context {
             scheduler.interrupt_recently_seen(contact_id, timestamp);
         }
     }
+
+    pub(crate) fn send_log_event(&self, event: EventType) -> anyhow::Result<()> {
+        if let Some(scheduler) = &*self.scheduler.try_read()? {
```

Comment:
It's a bit unfortunate that events will be lost if the scheduler is busy - why not inline `scheduler.send_debug_logging_event` and put the `debug_logging_send` Sender on Context

Reply:
As I'm rewriting this I noticed, that we probably still want to start that loop with `start_io`, so I would have to wrap the two values in options<> which is kinda ugly. Can still do it, but as it has the same lifetime as the scheduler it fits there. Maybe keep it that way and we refactore it if we *actually* miss events?